### DISCUSSION TITLE
Garnet synthetic traffic - Standalone protocol - Output Result is weird

### DISCUSSION THREAD

**Discussion by free-zxy** [gem5/gem5#1599] [Q&A]
Dear gem5 Community,

My experimental environment is Ubuntu 24.04 and the version of Gem5 is 24.0.0.1. 

Configure and compile according to the instructions on the gem5 official website. The link is as follows.

[gem5: Building gem5](https://www.gem5.org/documentation/general_docs/building#docker)

[gem5: Garnet Synthetic Traffic](https://www.gem5.org/documentation/general_docs/ruby/garnet_synthetic_traffic/)

Then start a test run with command :
./build/X86/gem5.opt configs/example/garnet_synth_traffic.py --num-cpus=16 --num-dirs=16 --network=garnet --topology=Mesh_XY --mesh-rows=4 --sim-cycles=1000 --synthetic=uniform_random --injectionrate=0.01

The m5out/stats.txt file outputs the following.

---------- Begin Simulation Statistics ----------
simSeconds                                   0.000000                       # Number of seconds simulated (Second)
simTicks                                         1000                       # Number of ticks simulated (Tick)
finalTick                                        1000                       # Number of ticks from beginning of simulation (restored from checkpoints and never reset) (Tick)
simFreq                                  1000000000000                       # The number of ticks per simulated second ((Tick/Second))
hostSeconds                                      0.16                       # Real time elapsed on the host (Second)
hostTickRate                                     6378                       # The number of ticks simulated per host second (ticks/s) ((Tick/Second))
hostMemory                                     784024                       # Number of bytes of host memory used (Byte)
system.clk_domain.clock                          1000                       # Clock period in ticks (Tick)
.

.

.

system.ruby.network.average_flit_latency          nan                       (Unspecified)
system.ruby.network.average_flit_network_latency          nan                       (Unspecified)
system.ruby.network.average_flit_queueing_latency          nan                       (Unspecified)
system.ruby.network.average_flit_vnet_latency |         nan                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.average_flit_vqueue_latency |         nan                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.average_hops                  nan                       (Unspecified)
system.ruby.network.average_packet_latency          nan                       (Unspecified)
system.ruby.network.average_packet_network_latency          nan                       (Unspecified)
system.ruby.network.average_packet_queueing_latency          nan                       (Unspecified)
system.ruby.network.average_packet_vnet_latency |         nan                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.average_packet_vqueue_latency |         nan                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.avg_link_utilization            0                       (Unspecified)

Secondly, I modified the command line as follows.

./build/X86/gem5.opt configs/example/garnet_synth_traffic.py --num-cpus=16 --num-dirs=16 --network=garnet --topology=Mesh_XY --mesh-rows=4 --sim-cycles=10000 --synthetic=uniform_random --vcs-per-vnet=16 --inj-vnet=0 --routing-algorithm=0 --injectionrate=0.02 --router-latency=4 --link-latency=1

The m5out/stats.txt file outputs the following.
---------- Begin Simulation Statistics ----------
simSeconds                                   0.000000                       # Number of seconds simulated (Second)
simTicks                                        10000                       # Number of ticks simulated (Tick)
finalTick                                       10000                       # Number of ticks from beginning of simulation (restored from checkpoints and never reset) (Tick)
simFreq                                  1000000000000                       # The number of ticks per simulated second ((Tick/Second))
hostSeconds                                      0.17                       # Real time elapsed on the host (Second)
hostTickRate                                    57175                       # The number of ticks simulated per host second (ticks/s) ((Tick/Second))
hostMemory                                     788120                       # Number of bytes of host memory used (Byte)
system.clk_domain.clock                          1000                       # Clock period in ticks (Tick)
.

.

.

system.ruby.network.average_flit_latency         9000                       (Unspecified)
system.ruby.network.average_flit_network_latency         8000                       (Unspecified)
system.ruby.network.average_flit_queueing_latency         1000                       (Unspecified)
system.ruby.network.average_flit_vnet_latency |        8000                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.average_flit_vqueue_latency |        1000                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.average_hops                    2                       (Unspecified)
system.ruby.network.average_packet_latency         9000                       (Unspecified)
system.ruby.network.average_packet_network_latency         8000                       (Unspecified)
system.ruby.network.average_packet_queueing_latency         1000                       (Unspecified)
system.ruby.network.average_packet_vnet_latency |        8000                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.average_packet_vqueue_latency |        1000                       |         nan                       |         nan                       (Unspecified)
system.ruby.network.avg_link_utilization     0.200000                       (Unspecified)

This output is very strange. I used gem5 V20.0.0 before, and there was no such strange result. 

Can anyone help me solve this confusion? Thanks!

Best regards！

**Comment by powerjg** [comment]
I'm not super familiar with garnet or the standalone protocol... could you explain what you think is strange? What is the expected results?

**Comment by free-zxy** [comment]
When injectionrate is 0.01，average_flit_latency is nan.
When injectionrate is 0.02, average_flit_latency is 9000.
The expected result is not large.

**Reply by powerjg** [reply]
Maybe you're not running long enough. 10,000 cycles with a 1% injection rate is ~100 packets (assuming the cycle time and injection rate multiplier are the same). I'm not familiar (you should carefully read the code to find out), but it's possible that `sim-cycles` is actually in Ticks, so you may be only simulating 10ns. 

Try simulating for millions or billions of "cycles"

**Reply by free-zxy** [reply]
Thanks, I have tried simulating for millions or billions of "cycles". But the average_flit_latency also is large. In my experience, this result is incorrect. Because I used before the Gem5 v20, the output result was inconsistent with this result.
